### PR TITLE
[DS-2841] REST API: Repaired pagination in get collections method.

### PR DESCRIPTION
**Pull request by ctu-developers**
Get collections method has bad logic of pagination for items and collections in list.

https://jira.duraspace.org/browse/DS-2841

### CODE REVIEW

File: dspace-rest/src/main/java/org/dspace/rest/CollectionsResource.java [hunk 1]
```diff
@@ -97,8 +97,8 @@
     @Path("/{collection_id}")
     @Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
     public org.dspace.rest.common.Collection getCollection(@PathParam("collection_id") String collectionId,
-            @QueryParam("expand") String expand, @QueryParam("limit") @DefaultValue("100") Integer limit,
-            @QueryParam("offset") @DefaultValue("0") Integer offset, @QueryParam("userIP") String user_ip,
+            @QueryParam("expand") String expand, @QueryParam("itemsLimit") @DefaultValue("100") Integer itemsLimit,
+            @QueryParam("itemsOffset") @DefaultValue("0") Integer itemsOffset, @QueryParam("userIP") String user_ip,
```
Comment (peterdietz) [inline dspace-rest/src/main/java/org/dspace/rest/CollectionsResource.java]: My only concern is that changing this @QueryParam, we would be renaming /collections/{ID}?limit=10&offset=10 to /collections/{ID?itemsLimit=10&itemsOffset=10, causing an incompatibility.

So, perhaps it is fine to change the javadoc, and the variable name, but leave the existing QueryParam value.

Comment (ctu-developers) [inline dspace-rest/src/main/java/org/dspace/rest/CollectionsResource.java]: Yes, i agree.

**Comment by peterdietz** [comment]
+1 This is good to go now

**Comment by tdonohue** [comment]
This needs refactoring as it has conflicts with current "master"

**Comment by tdonohue** [comment]
Thanks for your Pull Request. However, I'm closing this PR as it conflicts with our upcoming DSpace 7 REST API work (taking place currently on our `rest7` feature branch, and moving over to `master` soon). 

This PR can still move forward through one of the following routes:
1. You can choose to rework this PR to align with the new 7.x REST API efforts. Keep in mind, this will be a rebuilt REST API, and not all features have been completed/rebuilt. But, we welcome other collaborators to help move the effort forward. See https://wiki.duraspace.org/display/DSPACE/DSpace+7+UI+Working+Group
2. Or, you can resubmit this PR to our `dspace-6_x` maintenance branch as a simple patch to our 6.x REST API.

Again, thanks for the work. We hope you consider joining us in the DSpace 7 REST API efforts, but we also appreciate any patches to the existing 6.x REST API you are willing to provide.

**Pull request closed without merging**